returnn/torch/engine.py

```diff
@@ -359,9 +438,24 @@ def _run_step(self, extern_data_raw: Dict[str, torch.Tensor], *, train_flag: boo
 
         rf.init_train_step_run_ctx(train_flag=train_flag)
 
-        with autocast(device_type=self._device, dtype=self._autocast_dtype) if self._use_autocast else nullcontext():
-            sentinel_kw = {"__fwd_compatible_random_arg_%i" % int(random() * 100): None}
-            self._train_step_func(model=self._orig_model, extern_data=extern_data, **sentinel_kw)
+        sentinel_kw = {"__fwd_compatible_random_arg_%i" % int(random() * 100): None}
+
+        with autocast(
+            device_type=self._device, dtype=self._autocast_dtype
+        ) if self._use_autocast else nullcontext():
+            if not self._use_torch_distributed:
```

Comment:
There should be no if-statement here. You always should call the same `_train_step_func` func, in the same way.

The if-statement should be for the context manager, so inside the `with` statement above.